src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1551,14 +1590,30 @@ module T = struct
           let res = try_with_coinbase () in
           make_diff res None
 
+  let can_apply_supercharged_coinbase_exn ~delegator ~self ~epoch_ledger
+      ~global_slot =
+    let delegatee_unlocked =
+      Sparse_ledger.has_locked_tokens_exn ~global_slot
+        ~account_id:(Account_id.create self Token_id.default)
+        epoch_ledger
+      |> not
+    in
+    if Public_key.Compressed.equal self delegator || delegatee_unlocked then
```

Comment:
Should this not be `|| not delegatee_unlocked`?

Reply:
I believe this is correct -- @es92 ?

Reply:
This doesn't match the logic in proof_of_stake.ml above, and the implementation there (both need to be unlocked) makes more sense to me. Will comment the line with the discrepancy.

Reply:
Whats important is a locked account can't just get supercharged rewards by delegating to an unlocked account. Lmk if that helps, I haven't dug into the code but can do if it would be helpful

Reply:
I was looking at the issue #5753 again and I think I might have gotten confused with the example.  The condition should basically be- apply supercharged coinbase if the winner does not have any locked tokens and doesn't matter if the account it is delegating to has locked tokens or not. Is that how you read that example or am I confused again? @imeckler @mrmr1993

Reply:
Here `delegator` is the winner and so I should just be checking if `delegator` has any locked tokens

Reply:
Oops, didn't see the comment by @es92 before replying. But I think we are saying the same thing